apps/src/templates/manageStudents/ManageStudents.jsx

```diff
@@ -3,33 +3,89 @@ import PropTypes from 'prop-types';
 import {connect} from 'react-redux';
 import ManageStudentsTable from './ManageStudentsTable';
 import SyncOmniAuthSectionControl from '@cdo/apps/lib/ui/SyncOmniAuthSectionControl';
+import $ from 'jquery';
+import {
+  convertStudentServerData,
+  setStudents
+} from '@cdo/apps/templates/manageStudents/manageStudentsRedux';
+import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
 
 class ManageStudents extends React.Component {
   static propTypes = {
     studioUrlPrefix: PropTypes.string,
     pegasusUrlPrefix: PropTypes.string,
 
     // Provided by redux
-    sectionId: PropTypes.number
+    sectionId: PropTypes.number,
+    loginType: PropTypes.string,
+    setStudents: PropTypes.func.isRequired,
+    studentData: PropTypes.object
   };
 
+  state = {
+    isLoading: true
+  };
+
+  componentDidMount() {
+    // Don't load data if it's already stored in redux.
+    const alreadyHaveStudentData =
+      Object.values(this.props.studentData)[0] &&
+      Object.values(this.props.studentData)[0].sectionId ===
+        this.props.sectionId;
+    if (!alreadyHaveStudentData) {
+      $.ajax({
+        method: 'GET',
+        url: `/dashboardapi/sections/${this.props.sectionId}/students`,
+        dataType: 'json'
+      }).done(studentData => {
+        const convertedStudentData = convertStudentServerData(
+          studentData,
+          this.props.loginType,
+          this.props.sectionId
+        );
+        this.props.setStudents(convertedStudentData);
```

Comment:
Also nonblocking: Curious how others feel about this, since the data ends up in redux I'd advocate for the redux module owning the async request / convert / store process for this data and expose less of that implementation logic to the component here.

Reply:
> I'd advocate for the redux module owning the async request / convert / store process for this data and expose less of that implementation logic to the component here.

yes, I agree with this as a best practice, and also that it should be non-blocking for this PR since this code is being moved rather than added or updated.

Reply:
implemented: https://github.com/code-dot-org/code-dot-org/pull/31278